controllers/match.js

```diff
@@ -0,0 +1,72 @@
+const MatchData = require("../models/match");
+const { getDateRange, getYesterdayDate } = require("../utils/dateUtils");
+const { postData } = require('./postData');
+const scrapeController = require("./webscrapeController");
+
+const yesterdayDate = getYesterdayDate();
+
+exports.createMatch = (req, res) => {
+  const matchesData = req.body.matches;
+
+  const matchData = new MatchData({
+    matches: matchesData.map((match) => ({
+      matchDate: match.matchDate,
+      league: match.league,
+      homeTeam: match.homeTeam,
+      awayTeam: match.awayTeam,
+      homeScore: match.homeScore,
+      awayScore: match.awayScore,
+      matchStatus: match.matchStatus,
+      homeLogo: match.homeLogo,
+      awayLogo: match.awayLogo,
+    })),
+  });
+
+  matchData
+    .save()
+    .then((result) => {
+      res.status(201).json({
+        message: "Match data added successfully",
+        matches: result.matches,
+      });
+    })
+    .catch((err) => {
+      res.status(500).json({
+        message: "Failed to add match data",
+        error: err,
+      });
+    });
+};
+
+exports.getPastResults = async (_, res) => {
+  try {
+    const allMatchData = await scrapeController.scrapeEspn(yesterdayDate, yesterdayDate);
+    const postSuccess = await postData(allMatchData);
```

Comment:
put whatever you wrote in postData here instead of separating it into different files